app/src/main/java/org/wikipedia/dataclient/RestService.kt

```diff
@@ -82,16 +82,20 @@ interface RestService {
     fun getRelatedPages(@Path("title") title: String?): Observable<RbRelatedPages>
 
     @GET("page/media-list/{title}/{revision}")
-    fun getMediaList(
+    suspend fun getMediaListSuspend(
         @Path("title") title: String,
         @Path("revision") revision: Long
-    ): Observable<MediaList>
+    ): MediaList
 
+    /**
+     * Support method added to pass GalleryClientTest
+     * TODO: To be demised
+     */
     @GET("page/media-list/{title}/{revision}")
-    suspend fun getMediaListSuspend(
+    fun getMediaList(
         @Path("title") title: String,
         @Path("revision") revision: Long
-    ): MediaList
+    ): Observable<MediaList>
```

Comment:
Currently there is one use of `getMediaList` which is in `GalleryClientTest`, I do not know if this method is required in future.

But `getMediaListSuspend` is already being used `LinkPreviewViewModel`, and I have used in `GalleryRepository`.

Considering this i have added demise comment, so that devs can demise when `GalleryClientTest ` starts using `getMediaListSuspend `. 

Do let me know if you think these needs to be removed/changed.

Reply:
From what I see, the `getMediaList` in line `85` has been changed to line `95` which the changes you made were swapping two functions.